src/leo_gateway_http_commons.erl

```diff
@@ -333,8 +333,16 @@ get_object_with_cache(Req, Key, CacheObj, #req_params{bucket                 = B
             {ok, CustomHeaders} = leo_nginx_conf_parser:get_custom_headers(Key, CustomHeaderSettings),
             Headers2 = Headers ++ CustomHeaders,
             BodyFunc = fun(Socket, _Transport) ->
-                               file:sendfile(CacheObj#cache.file_path, Socket, 0, 0, [{chunk_size, SendChunkLen}]),
-                               ok
+                               case file:open(CacheObj#cache.file_path, [raw, read]) of
```

Comment:
I recommend you use [filelib:is_file/1](http://erlang.org/doc/man/filelib.html#is_file-1) before calling ``file:open/2``.